newa/cli.py

```diff
@@ -231,30 +261,87 @@ def cmd_schedule(ctx: CLIContext) -> None:
 
         # create few fake Issue objects for now
-        for request in config.build_requests():
-            request_job = RequestJob(
+        for request in requests:
+            schedule_job = ScheduleJob(
                 event=jira_job.event,
                 erratum=jira_job.erratum,
                 jira=jira_job.jira,
                 recipe=jira_job.recipe,
                 request=request)
-            ctx.save_request_job('request-', request_job)
+            ctx.save_schedule_job('schedule-', schedule_job)
 
 
 @main.command(name='execute')
+@click.option(
+    '--workers',
+    default=4,
+    )
 @click.pass_obj
-def cmd_execute(ctx: CLIContext) -> None:
+def cmd_execute(ctx: CLIContext, workers: int) -> None:
     ctx.enter_command('execute')
 
-    for erratum_job in ctx.load_erratum_jobs('schedule-'):
-        # worker = new Executor(yaml)
-        # run() returns result object
-        # result = worker.run()
+    if not os.environ.get("TESTING_FARM_API_TOKEN"):
+        raise ValueError("TESTING_FARM_API_TOKEN not set!")
+
+    # get a list of files to be scheduled so that they can be distributed across workers
+    schedule_list = [
+        os.path.join(
+            ctx.state_dirpath,
+            child.name)
+        for child in ctx.state_dirpath.iterdir()
+        if child.name.startswith('schedule-')]
+
+    worker_pool = multiprocessing.Pool(workers)
+    for _ in worker_pool.map(worker, schedule_list):
+        # small sleep to avoid race conditions inside tmt code
+        time.sleep(0.1)
+
+    print('Done')
+
+
+def worker(schedule_file: str) -> None:
+    workdir, basename = os.path.split(schedule_file)
+
+    print(f'{basename} processing request...')
+    # read request details
+    schedule_job = ScheduleJob.from_yaml_file(Path(schedule_file))
+    print(f'{basename} initiating TF request')
+    tf_request = schedule_job.request.initiate_tf_request()
+    # tf_request = TFRequest(
+    #    api='https://api.dev.testing-farm.io/v0.1/requests/519f5c01-46b6-47c9-a055-aecaa32e6a20',
+    #    uuid='519f5c01-46b6-47c9-a055-aecaa32e6a20')
+    print(f'{basename} TF request filed with uuid {tf_request.uuid}')
+    finished = False
+    delay = 60
+    while not finished:
+        time.sleep(delay)
+        tf_request.fetch_details()
+        state = tf_request.details['state']
+        print(f'{basename} TF reqest {tf_request.uuid} {state}')
+        finished = state == 'complete'
+    execute_job = ExecuteJob(
+        event=schedule_job.event,
+        erratum=schedule_job.erratum,
+        jira=schedule_job.jira,
+        recipe=schedule_job.recipe,
+        request=schedule_job.request,
+        execution=Execution(return_code=0, artifacts=tf_request.details['run']['artifacts']),
+        )
 
-        ctx.save_erratum_job('execute-', erratum_job)
+    name = basename.replace('schedule-', 'execute-')
+    output_file = os.path.join(workdir, name)
+    execute_job.to_yaml_file(Path(output_file))
```

Comment:
Can be simplified, thanks to `pathlib` magic (`Path / str` yield `Path`), to:

```python
execute_job.to_yaml_file(workdir / basename.replace('schedule-', 'execute-'))
```

Reply:
fixed